src/drivers/interrupt.c

```diff
@@ -27,6 +27,17 @@ DECLARE_SOF_UUID("irq", irq_uuid, 0x1862d39a, 0x3a84, 0x4d64,
 
 DECLARE_TR_CTX(irq_tr, SOF_UUID(irq_uuid), LOG_LEVEL_INFO);
 
+/* For i.MX with Zephyr make any second level
+ * interrupt handling go through interrupt-irqsteer.c
+ */
+#if defined(__ZEPHYR__) && defined(CONFIG_IMX)
+#define interrupt_get_irq mux_interrupt_get_irq
+#define interrupt_register mux_interrupt_register
+#define interrupt_unregister mux_interrupt_unregister
+#define interrupt_enable mux_interrupt_enable
+#define interrupt_disable mux_interrupt_disable
```

Comment:
Can we have this in the wrapper.c - This way it's easier to see what is left to move to Zephyr native APIs etc, I'm pretty sure the irqsteer logic will need moved into Zephyr too

Reply:
> Can we have this in the `wrapper.c` - This way it's easier to see what is left to move to Zephyr native APIs etc,

I cannot move this in wrapper.c. The idea is to _not_ use the functions (`interrupt_register()`, `interrupt_unregister()`, `interrupt_enable()`, etc.) from `wrapper.c` or `interrupt.c`, these should go through `interrupt-irqsteer.c`.

> I'm pretty sure the irqsteer logic will need moved into Zephyr too

Yes, I've also wrote this in the commit message: 
```
TODO: Implement a driver for the IRQ_STEER in Zephyr,
to replace the legacy code (interrupt-irqsteer.c).
```

Reply:
Ah, ok got you - so this is in your plan and you will use the Zephyr version of irq-steer once it's upstream.